debian/control

```diff
@@ -3,7 +3,8 @@ Priority: optional
 Maintainer: Nick Black <dankamongmen@gmail.com>
 Build-Depends: debhelper-compat (= 12), cmake (>= 3.13), pkg-config (>= 0.29),
  libncurses-dev (>= 6.1), libavformat-dev (>= 57.0), libswscale-dev (>= 5.0),
- libavutil-dev (>= 56.0), pandoc (>= 1.19.2.4), python3-all-dev (>= 3.2)
+ libavutil-dev (>= 56.0), pandoc (>= 1.19.2.4), python3-all-dev (>= 3.2),
+ g++ (>= 4:7.0.0-0)
```

Comment:
`build-essential` depends on `g++`, debian policy 4.2 prohibits explicit deps on dependencies of `build-essential`. in unstable at least that means we get 8.3+.

Reply:
I need to be sure we depend on gcc with C++17 (7.0+), though (boost depends on g++ too)

Reply:
Debian policy doesn't forbid this, [just checked](https://www.debian.org/doc/debian-policy/ch-source.html#package-relationships):

```
For example, if building a package requires a certain compiler, then the compiler should be specified as a build-time dependency.

It is not necessary to explicitly specify build-time relationships on a minimal set of packages that are always needed to compile, link and put in 
a Debian package a standard “Hello World!” program written in C or C++. The required packages are called build-essential, and an informational 
list can be found in /usr/share/doc/build-essential/list (which is contained in the build-essential package).
```

Reply:
Ahhh so this is for a particular *revision* of the compiler, got it. In that case, should it not (ideally) be an OR dependency between either g++ 7.0 or clang 5?. don't need to change it now, just wondering.

Reply:
I think it should depend on the "default" compiler used by the OS, so in this case it would be g++. Clang **should** of course work but it's no The Compiler for Debian/Ubuntu and there might be some issues when mixing other libs with notcurses++ - if one is built with clang++ and another with g++ there's a potential for trouble (even though clang now uses libstdc++ by default). Better safe than sorry I guess.